### PR TITLE
ISPN-10417 Kerberos authentication for the server

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-10417

This PR has a complete server-side implementation of GSSAPI and GS2-KRB5 for Hot Rod and SPNEGO for HTTP.

### CODE REVIEW

**Comment by tristantarrant** [comment]
I have added a SPNEGO authenticator to our rest client. This is now complete

**Comment by tristantarrant** [comment]
Failures are unrelated

File: server/runtime/src/test/java/org/infinispan/server/configuration/JsonSerializationTest.java [hunk 1]
```diff
@@ -125,6 +129,13 @@ public void testJsonSerialization() throws Exception {
       assertEquals("ldap://${org.infinispan.test.host.address}:10389", ldapRealm.get("url").asText());
       assertEquals("uid=admin,ou=People,dc=infinispan,dc=org", ldapRealm.get("principal").asText());
       assertEquals("***", ldapRealm.get("credential").asText());
+
+      /*JsonNode ldapNameRewriter = ldapRealm.get("name-rewriter");
+      JsonNode ldapRegexPrincipalTransformer = ldapNameRewriter.get("regex-principal-transformer");
+      assertEquals("uid", ldapRegexPrincipalTransformer.get("name").asText());
+      assertEquals("uid", ldapRegexPrincipalTransformer.get("pattern").asText());
+      assertEquals("uid", ldapRegexPrincipalTransformer.get("replacement").asText());*/
+
```
Comment (gustavonalle) [inline server/runtime/src/test/java/org/infinispan/server/configuration/JsonSerializationTest.java]: Is this part not working?

Comment (gustavonalle) [reply server/runtime/src/test/java/org/infinispan/server/configuration/JsonSerializationTest.java]: @tristantarrant Could you create a JIRA detailing the parts of the config that are not being serialized at the moment?

Comment (tristantarrant) [reply server/runtime/src/test/java/org/infinispan/server/configuration/JsonSerializationTest.java]: https://issues.jboss.org/browse/ISPN-10961

**Comment by gustavonalle** [comment]
Merged, thanks @tristantarrant

**Pull request merged**